satellite/satellitedb/overlaycache.go

```diff
@@ -36,6 +36,25 @@ type overlaycache struct {
 	db *satelliteDB
 }
 
+func (cache *overlaycache) GetAllNodeIDs(ctx context.Context) (_ []storj.NodeID, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	dbxNodeIDs, err := cache.db.All_Node_Id(ctx)
+	if err != nil {
+		return nil, err
+	}
+	nodeIDs := []storj.NodeID{}
+	for _, id := range dbxNodeIDs {
+		nID, err := storj.NodeIDFromBytes(id.Id)
+		if err != nil {
+			continue
```

Comment:
Should we add the error to an error group here?`
Else the return statement hasnt much benefit of returning the error at all.

Reply:
If this error is happening, I'd prefer some sort of hard failure mode happens. This would mean both that the database has invalid weird data in it and that specific node would never get paid.